subsystems/armSubsystem.py

```diff
@@ -0,0 +1,126 @@
+    #     wpilib.SmartDashboard.putNumber("I Zone", self.kIz)
+    #     wpilib.SmartDashboard.putNumber("Feed Forward", self.kFF)
+    #     wpilib.SmartDashboard.putNumber("Max Output", self.kMaxOutput)
+    #     wpilib.SmartDashboard.putNumber("Min Output", self.kMinOutput)
+    #     wpilib.SmartDashboard.putNumber("Set Rotations", 0)
+        
+        
+    # def periodic(self) -> None:
+        
+    #     # Read PID coefficients from SmartDashboard
+    #     p = wpilib.SmartDashboard.getNumber("P Gain", 0)
+    #     i = wpilib.SmartDashboard.getNumber("I Gain", 0)
+    #     d = wpilib.SmartDashboard.getNumber("D Gain", 0)
+    #     iz = wpilib.SmartDashboard.getNumber("I Zone", 0)
+    #     ff = wpilib.SmartDashboard.getNumber("Feed Forward", 0)
+    #     max = wpilib.SmartDashboard.getNumber("Max Output", 0)
+    #     min = wpilib.SmartDashboard.getNumber("Min Output", 0)
+    #     rotations = wpilib.SmartDashboard.getNumber("Set Rotations", 0)
+        
+    #     # If PID coefficients on SmartDashboard have changed, write new values to controller
+    #     if (p != self.kP): self.pidArm1.setP(p); self.pidArm2.setP(p); self.kP = p
+    #     if (i != self.kI): self.pidArm1.setI(i); self.pidArm2.setI(i); self.kI = i
+    #     if (d != self.kD): self.pidArm1.setD(d); self.pidArm2.setD(i); self.kD = d
+    #     if (iz != self.kIz): self.pidArm1.setIZone(iz); self.pidArm2.setIZone(iz); self.kIz = iz
+    #     if (i != self.kI): self.pidArm1.setFF(ff); self.pidArm2.setFF(ff); self.kFF = ff
+    #     if ((max != self.kMaxOutput) and (min != self.kMinOutput)): 
+    #         self.pidArm1.setOutputRange(min, max); self.pidArm2.setOutputRange(min, max); self.kMaxOutput = max; self.kMinOutput = min
+            
+    #     # Set desired arm position
+    #     self.pidArm1.setReference(rotations, rev.CANSparkMax.ControlType.kPosition)
+    #     self.pidArm2.setReference(rotations, rev.CANSparkMax.ControlType.kPosition)
+        
+    #     wpilib.SmartDashboard.putNumber("SetPoint", rotations)
+    #     wpilib.SmartDashboard.putNumber("Motor 1 Position", self.encoder1.getPosition())
+    #     wpilib.SmartDashboard.putNumber("Motor 2 position", self.encoder2.getPosition())
+    #     # wpilib.SmartDashboard.putNumber("tof Sensor Range in millimeters", self.tofSensor.getRange())
+        
+    def armwithjoystick(self, joystickInput):
+        if joystickInput <= 0.15 and joystickInput >= -0.15:
+            calculatedInput = 0
+        else:
+            calculatedInput = joystickInput
+            
+        if self.armLimitSwitch.get():
+            speed = (calculatedInput * constants.SW.ArmSpeed)
+        else: 
+            if calculatedInput > 0:
+                speed = 0
+            else:
+                speed = (calculatedInput * constants.SW.ArmSpeed)
+            
+        self.motorgroup.set(speed)
+
+    def arm_down(self, speed):
+        self.motorgroup.set(speed)
```

Comment:
If the limit switch is active, the arm shouldn't even try to move down.  (However, if the limit switch isn't *yet* active and we start moving the arm down and that activates the limit switch soon, we can't do anything about that here in the subsystem -- we need to use a Command for that.)